lib/screens_web/controllers/v2/screen_controller.ex

```diff
@@ -118,7 +118,25 @@ defmodule ScreensWeb.V2.ScreenController do
     render_not_found(conn)
   end
 
-  def widget(conn, %{"id" => app_id, "widget" => widget_data})
+  # Handles widget page GET requests with widget data as a query param.
+  # Phoenix does not automatically decode JSON received in query params.
+  def widget(conn, %{"app_id" => app_id, "widget" => json_data}) when is_binary(json_data) do
```

Comment:
Why do we need both a GET and a POST? Should I have used a GET to begin with?

Reply:
Good question honestly, I'm not sure. GET _is_ more standard for requesting a webpage, and with `encodeURI` it's not too difficult to plop the JSON into the URL.

The only real concern I can think of is about URL length. I only tested the GET endpoint with one of our simplest widgets, the e-ink footer. From some stackoverflow searching, it looks like the maximum URL length is around 8000 characters, so I think we'd be fine, but it's still a slight concern.

I'd be happy to remove the POST endpoint if you think that makes sense!
Or we could do something like have them load our widget page in an iframe, and they supply the widget data strictly client-side. This would be similar to how we're able to get OFM tags using `parent.mraid`. We could do `parent.widgetData`. We're probably past the point where it's reasonable to consider new ideas, though—the current approach is more than good enough.

Reply:
Yeah, that works.